website/templates/profile/oauth_app_detail.mako

```diff
@@ -0,0 +1,119 @@
+                        <p data-bind="validationMessage: name, visible: $root.showMessages()" class="text-danger"></p>
+                    </div>
+
+                    <div class="form-group">
+                        <label>Project homepage URL</label>
+                        <input class="form-control" type="url" data-bind="value: homeUrl" required="required" placeholder="Required">
+                        <p data-bind="validationMessage: homeUrl, visible: $root.showMessages()" class="text-danger"></p>
+                    </div>
+
+                    <div class="form-group">
+                        <label>Application description</label>
+                        <textarea class="form-control" placeholder="Optional" data-bind="value: description"></textarea>
+                        <p data-bind="validationMessage: description, visible: $root.showMessages()" class="text-danger"></p>
+                    </div>
+
+                    <div class="form-group">
+                        <label>Authorization callback URL</label>
+                        <input type="url" class="form-control" data-bind="value: callbackUrl" required="required" placeholder="Required">
+                        <p data-bind="validationMessage: callbackUrl, visible: $root.showMessages()" class="text-danger"></p>
+                    </div>
+
+                    <!-- Flashed Messages -->
+                    <div class="help-block">
+                        <p data-bind="html: $root.message, attr.class: $root.messageClass"></p>
+                    </div>
+
+                    <div class="padded">
+                        <button type="reset" class="btn btn-default"
+                                data-bind="click: $root.cancelChange.bind($root)">Cancel</button>
+                        <button type="submit" class="btn btn-success"
+                                data-bind="visible: $root.isCreateView()">Create</button>
+                        <button type="submit" class="btn btn-success"
+                                data-bind="visible: !$root.isCreateView(), click: $root.updateApplication.bind($root), enable: $root.dirty()">Save</button>
+                    </div>
+                </form>
+            </div>
+        </div> <!-- End appDetail section -->
+    </div>
+</div>
+</%def>
+
+<%def name="javascript_bottom()">
+<script type="text/javascript">
+    window.contextVars = window.contextVars || {};
+    window.contextVars.urls = {
+        webListUrl: ${ web_url_for('oauth_application_list') | sjson, n },
+        apiListUrl: ${ app_list_url | sjson, n },
+        apiDetailUrl: ${ app_detail_url | sjson, n }
+    };
+
+    // Make sure to display tooltips correctly
+    $(document).ready(function(){
+        $('[data-toggle="tooltip"]').tooltip();
+    });
+
+</script>
+<script src=${"/static/public/js/profile-settings-applications-detail-page.js" | webpack_asset}></script>
```

Comment:
Same as above: two view functions use this template (create + edit). Both already have the dev_only decorator.